lib/utils/group-helper.js

```diff
@@ -115,6 +115,29 @@ class GroupHelper {
         if (filename && Array.isArray(filename)) {
             return filename.map(fp => this.resolveFilePath(fp, defaultValue)).filter(e => e);
         }
+        if (filename) {
+            if (!defaultValue) {
+                const fileNameArray = filename.split('/');
+                defaultValue = fileNameArray[fileNameArray.length - 1]
+                const isLastNameAFile = existsSync(
+                    resolve(
+                        join(
+                            fileNameArray.slice(0, fileNameArray.length - 1).join('/'),
+                            defaultValue.includes('.js') ? defaultValue : defaultValue + ".js"
+                        )
+                    )
+                );
+                defaultValue = isLastNameAFile ? defaultValue : 'index.js';
+                defaultValue = defaultValue.includes('.js') ? defaultValue : defaultValue + ".js"
+                filename = (defaultValue == 'index.js' && (fileNameArray[fileNameArray.length - 1] != 'index' && fileNameArray[fileNameArray.length - 1] != 'index.js')) ? fileNameArray.join('/') : fileNameArray.slice(0, fileNameArray.length - 1).join('/')
+            }
```

Comment:
All these chained conditionals makes the code block a bit hard to read :thinking:

Reply:
yeah kind of, do you have any better refactoring in mind!? would love to get some help then !

Reply:
It's really hard to understand what you try to achieve in the new code. Why don't you make a separate function and try to use the old fashion if/else?

Reply:
like to extract everything inside this if statement to a separate function? or just this line Line number 132 ?

Reply:
All of it. Having a function with a proper name would help to understand what we try to achieve.

Reply:
done